cardano-api/src/Cardano/Api/MetaData/Pool.hs

```diff
@@ -0,0 +1,122 @@
+module Cardano.Api.MetaData.Pool
+  ( StakePoolMetadata
+  , StakePoolMetadataValidationError (..)
+  , decodeAndValidateStakePoolMetadata
+  , renderStakePoolMetadataValidationError
+  ) where
+
+import           Cardano.Prelude
+import           Prelude (String)
+
+import           Control.Monad.Fail (fail)
+
+import           Data.Aeson (FromJSON (..), (.:))
+import qualified Data.Aeson as Aeson
+import qualified Data.Aeson.Types as Aeson
+import qualified Data.ByteString as BS
+import qualified Data.Text as Text
+
+-- | A representation of the required fields for stake pool metadata.
+data StakePoolMetadata = StakePoolMetadata
+  { _spmName :: !Text
+    -- ^ A name of up to 50 characters.
+  , _spmDescription :: !Text
+    -- ^ A description of up to 255 characters.
+  , _spmTicker :: !Text
+    -- ^ A ticker of 3-5 characters, for a compact display of stake pools in
+    -- a wallet.
+  , _spmHomepage :: !Text
```

Comment:
Are we missing the validation on the homepage URL size?
https://jira.iohk.io/browse/CAD-736?focusedCommentId=19365&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-19365